### DISCUSSION TITLE
What's stoping a user from skipping nonce in password change?

### DISCUSSION THREAD

**Discussion by bombillazo** [supabase/supabase#14857] [Questions] [answered]
So my question is, in the frontend the `updateUser` function can update the password with or without a nonce value, what is stopping an auth user from changing the password without using the nonce if they edit the client code to skip it? What if for security reasons we want to enforce nonce on all password changes?

**Comment by j4w8n** [comment, marked as answer]
In your Supabase dashboard > Authentication > Providers > Email, you'd enable `Secure password change`. With that enabled, a nonce is required otherwise the request fails.

**Reply by rajdtta** [reply]
Think this is bugged. I have `Secure password change` enabled, but I'm still able to change passwords without the correct nonce being used.

**Reply by j4w8n** [reply]
> Think this is bugged. I have `Secure password change` enabled, but I'm still able to change passwords without the correct nonce being used.

Can you open a separate issue for this? Will get tracked better.
